### DISCUSSION TITLE
Grid Point Retrieval (Inconsistent Time Steps)

### DISCUSSION THREAD

**Discussion by grfritz** [weather-gov/api#379] [Q&A]
I have been initially trying out the API grid point forecast feature (i.e. https://api.weather.gov/gridpoints/IND/57,68) and discovered the data (such as quantitativePrecipitation) is usually displayed, after the initial posted value, in 6 hour time steps.  Some through (such as https://api.weather.gov/gridpoints/SLC/99,174) have values posted as a mix of 3 hour and 6 hour time steps, even when there is no change in data values.  I was wondering whether this is normal.   If so,  I will need to update our software to accomodate.   This is a departure from the consistent 6 hour periods posted for same product on the NDFD.   Appreciate any feedback.

**Comment by StephenClouse** [comment]
This is normal behavior.

As noted in #20 we attempt to represent the raw data without modification. One of the few changes we _do_ make to the data is that we apply run-length encoding to reduce the size of the JSON output. For example, if we receive [70, 70, 70] as three consecutive hourly temperature data points, we will represent this as 70 with a 3-hour duration (PT3H).

quantitativePrecipitation is an exception to the above, as this data set does not represent instantaneous data points, but accumulation totals for each valid time period. Applying the RLE algorithm would change these semantics, so we leave this data set as is. And as you have noted, the raw data we receive generally does use this expanding window as the data gets further out in the forecast period.
